frontend/src/store/modules/profile/actions.js

```diff
@@ -0,0 +1,68 @@
+import httpService from '../../../services/common/httpService';
+
+export default {
+    setUser: (context, user) => {
+        context.commit('SET_USER', user);
+    },
+    updateProfile: (context, data) => {
+        return new Promise((resolve, reject) => {
+            httpService.post('/users/' + data.userId + '/profile', data.formData)
+                .then(function (res) {
+                    context.commit('SET_USER_AVATAR', res.data.data.avatar_url);
+                    context.dispatch('auth/fetchAuthenticatedUser', {}, {root: true});
```

Comment:
remove this dispatch. no need to query auth user, this api '/users/' + data.userId + '/profile can return updated data